qctrlopencontrols/dynamic_decoupling_sequences/driven_controls.py

```diff
@@ -227,7 +219,7 @@ def convert_dds_to_driven_control(
         half_pulse_duration  = 0.
 
         if not np.isclose(operations[1, op_idx], 0.): # Rabi rotation
-            half_pulse_duration = 0.5 * max(operations[1, op_idx] / maximum_rabi_rate,
+            half_pulse_duration = 0.5 * max(np.abs(operations[1, op_idx]) / maximum_rabi_rate,
```

Comment:
Assuming that previously if a negative rotation was provided we'd eventually throw an error (due to the segment ending before it started), I think we should continue to do that now (just with a more explicit check and error message). I'd be a bit scared taking an absolute value, since then we might be silently giving a user a result they didn't expect.

Reply:
Fair enough.

For what's worth, an error was still being thrown if the user selected a negative Rabi rotation, only further down the line, because DrivenControls doesn't accept negative Rabi rates. (Although the DDS class does accept negative Rabi rotations without throwing an error, which is a bit of a contradiction in my view.) In my opinion, the error that was being thrown (complaining explicitly about the wrong value of the Rabi rate) was more useful than the previous error (simply saying that something was wrong in creating a sequence and suggesting a change in the `maximum_rabi_rate`, which wouldn't help in this case).

It doesn't hurt to add an extra check, though, so now I added one that throws an exception before any Rabi rotation values are used

Reply:
Ah, OK, I didn't realise we'd throw an error anyway (and a useful error, at that). Still, in general I think it's best to throw as early as possible, and this way we still get to keep a useful error message.

Agree that it's inconsistent for DDS to accept negative Rabi rates -- maybe a future project could be to explicitly disallow them (or explicitly allow them, and make clear how they're interpreted). If that sounds reasonable I'll file a ticket.

Reply:
Sounds very reasonable to me!

Reply:
Filed https://q-ctrl.atlassian.net/browse/QENG-795. IMO not a high priority, so maybe we can start grouping together some "if we care about open controls then we should fix these" tickets.